src/tasks/publish-message-task.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  DatabaseTransactionHandler,
+  ItemMembershipService,
+  ItemService,
+  Member,
+} from 'graasp';
+import { ChatService } from '../db-service';
+import { ChatMessage } from '../interfaces/chat-message';
+import {
+  ItemNotFound,
+  MemberCannotReadItem,
+} from '../util/graasp-item-chat-error';
+import { BaseChatTask } from './base-chat-task';
+
+export class PublishMessageTask extends BaseChatTask<ChatMessage> {
+  get name(): string {
+    return PublishMessageTask.name;
+  }
+
+  constructor(
+    member: Member,
+    chatId: string,
+    data: Partial<ChatMessage>,
+    itemService: ItemService,
+    itemMembershipService: ItemMembershipService,
+    chatService: ChatService,
+  ) {
+    super(member, itemService, itemMembershipService, chatService);
+    this.targetId = chatId;
+    this.data = data;
+  }
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+
+    // get item for which we're fetching the chat
+    const item = await this.itemService.get(this.targetId, handler);
+    if (!item) throw new ItemNotFound(this.targetId);
```

Comment:
I prefer always wrapping if with {}. But willing to discuss.

Reply:
https://github.com/graasp/graasp-plugin-chatbox/pull/2#discussion_r672904328